packages/lodestar/src/chain/chain.ts

```diff
@@ -142,6 +142,8 @@ export class BeaconChain extends (EventEmitter as { new(): ChainEventEmitter })
 
     if(!await this.db.block.has(block.parentRoot)) {
       this.emit("unknownBlockRoot", block.parentRoot);
+      this.blockProcessingQueue.add(block);
+      return;
     }
 
     if(block.slot <= this.latestState.slot) {
```

Comment:
this condition can be changed to discarding if the block.slot is before the finalized block slot (according to fork choice)

We also probably want a check for if `this.db.block.has(blockHash)` already (duplicate block, already processed)